dashboard/app/views/levels/_external.html.haml

```diff
@@ -1,9 +1,15 @@
 %link{rel: 'stylesheet', href: asset_path('blockly/css/common.css')}
+- href = asset_path @level.properties['href']
+- is_markdown = href.end_with? '.md'
+= javascript_include_tag 'Showdown.min.js' if is_markdown
 
 #external
   :javascript
     var options = #{@level.properties['options'].to_json}
-
-    $.ajax("#{asset_path @level.properties['href']}").done(function (data) {
+    var href = "#{href}"
+    $.ajax(href).done(function (data) {
+      if(#{is_markdown}) {
+        data = new Showdown.converter().makeHtml(data);
```

Comment:
Though there's not much of a difference, the main factor was simplicity since we were already pulling in the html client-side through an ajax request anyway.

Setting up the conversion client-side also seemed like less things to maintain on the server, taking into account a long-term strategy to avoid tying 'apps' like these to too much server-side infrastructure to make them more easily migratable in the future. Since the conversion is just a string transformation, the integration was dead simple (just a single JavaScript file to add).